src/client/datascience/raw-kernel/liveshare/hostRawNotebookProvider.ts

```diff
@@ -127,14 +127,26 @@ export class HostRawNotebookProvider implements IRawNotebookProvider {
             // We need to locate kernelspec and possible interpreter for this launch based on resource and notebook metadata
             const displayName = getDisplayNameOrNameOfKernelConnection(kernelConnection);
 
-            const progressDisposable = !ui.disableUI
+            let progressDisposable = !ui.disableUI
```

Comment:
I'm wondering if the 'displayUi' object can just implement the progress indicator? It activates it when UI is true. 

Then you wouldn't need to put it in these places or have the event handler.

Reply:
Although maybe the event handler is still there, you used it to do something somewhere else besides progress.

Reply:
Good idea, let me try that.

Reply:
Nope, can't do that

Reply:
Wait, just thought of a way.

Reply:
@rchiodo I'd like to do this in debt week.
Basically have DisplayOptions have a method such as 
* `showProgress` which will return a disposable. 
    * If `disableUI === true `, then no progress is displayed & vice versa
	* When disposing the return value the progress is hidden (if displayed).
* Other parts of the code can call `showProgress` any time, then all we do is replace the message with the new one. & when that new message is disposed, we swap back to the old message (kinda like a stack)
* & we can have this displayOptions keyed by resource (or optionally add a resource argument to `showProgress`)
	* This way, when you start two notebooks they will have their own progress messages with their own messages
	
Thats the gist of the API, i believe this is pretty much what you too have in mind, however i'll need to refactor some code to make this work well & would prefer to do that in debt week.